suite/tests/linux/sigaction.c

```diff
@@ -207,9 +207,19 @@ test_sigprocmask()
     assert(make_sigprocmask(SIG_SETMASK + 1, &new, NULL, 8) == -1);
     assert(errno == EINVAL);
     /* Bad 'how' EINVAL gets reported before bad old sigset EFAULT. */
-    assert(make_sigprocmask(~0, &new, (void *)0x123, 8) == -1);
+    assert(make_sigprocmask(~0, &new, fault_addr, 8) == -1);
     assert(errno == EINVAL);
 
+    /* EFAULT due to bad old sigset still sets the new mask. */
```

Comment:
Really?  Wow: I thought in general syscalls should be single all-or-nothing transactions, and that a failure return value meant total failure, not some complex half-success.  This does not seem like a good thing for the kernel to do IMHO.  Please document this in the DR handling code because this is not what I would expect.

Reply:
Done.